lib/rspec/mocks/instance_method_stasher.rb

```diff
@@ -62,7 +62,11 @@ def method_owned_by_klass?
         # `#<MyClass:0x007fbb94e3cd10>`, rather than the expected `MyClass`.
         owner = owner.class unless Module === owner
 
-        owner == @klass || !(method_defined_on_klass?(owner))
+        owner == @klass ||
+          # When `extend self` is used, and not under any instance of
+          (owner.singleton_class == @klass &&
```

Comment:
Is this comment complete?

Reply:
Yes, by "any instance of" I meant `allow_any_instance_of` or `expect_any_instance_of`. I'll update it.